### PR TITLE
Allow passing password in URL for play.html

### PR DESCRIPTION
**Pull request by myrrc**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Allow `?password=pass` in URL. Password is replaced in browser history.

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
This is an automated comment for commit d033a5b66dd95af9250217c92f127c6be7da2851 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49505/d033a5b66dd95af9250217c92f127c6be7da2851/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: programs/server/play.html [hunk 1]
```diff
@@ -517,11 +517,12 @@
     let previous_query = '';
 
     const current_url = new URL(window.location);
+    const opened_locally = document.location.href.startsWith("file://");
```
Comment (alexey-milovidov) [inline programs/server/play.html:520]: Is location.protocol ok?

Comment (myrrc) [reply programs/server/play.html:520]: I'm not a JS expert but I believe checking href is same as checking location protocol.

File: programs/server/play.html [hunk 2]
```diff
@@ -532,6 +533,18 @@
         document.getElementById('user').value = user_from_url;
     }
 
+    const pass_from_url = current_url.searchParams.get('password');
+    if (pass_from_url) {
+        document.getElementById('password').value = pass_from_url;
+        if (!opened_locally) {
```
Comment (alexey-milovidov) [inline programs/server/play.html:539]: Why shouldn't we apply it if opened_locally?

Comment (myrrc) [reply programs/server/play.html:539]: Browsers don't allow manipulating history for a file:// protocol.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**